javascript-client/test/api/PartApi.spec.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Swagger Tunning Place
+ * This is a sample Tuning Palce server..
+ *
+ * OpenAPI spec version: 1.0.0
+ *
+ * NOTE: This class is auto generated by the swagger code generator program.
+ * https://github.com/swagger-api/swagger-codegen.git
+ *
+ * Swagger Codegen version: 2.4.6
+ *
+ * Do not edit the class manually.
+ *
+ */
+let expect = require("chai").expect;
+(function (root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    // AMD.
+    define(['expect.js', '../../src/index'], factory);
+  } else if (typeof module === 'object' && module.exports) {
+    // CommonJS-like environments that support module.exports, like Node.
+    factory(require('expect.js'), require('../../src/index'));
+  } else {
+    // Browser globals (root is window)
+    factory(root.expect, root.SwaggerTunningPlace);
+  }
+}(this, function (expect, SwaggerTunningPlace) {
+  'use strict';
+
+  var instance;
+
+  beforeEach(function () {
+    instance = new SwaggerTunningPlace.PartApi();
+    instance.apiClient.basePath = 'http://localhost:5000';
+  });
+
+  var getProperty = function (object, getter, property) {
+    // Use getter method if present; otherwise, get the property directly.
+    if (typeof object[getter] === 'function')
+      return object[getter]();
+    else
+      return object[property];
+  }
+
+  var setProperty = function (object, setter, property, value) {
+    // Use setter method if present; otherwise, set the property directly.
+    if (typeof object[setter] === 'function')
+      object[setter](value);
+    else
+      object[property] = value;
+  }
+
+  describe('PartApi', function () {
+    // describe('addPart', function () {
+    //   it('should call addPart successfully', function (done) {
+    //     let part = {
+    //       "title": "test title1",
+    //       "model": "test model1",
+    //       "year": 2018,
+    //       "description": "This is test description. All-in-one kit",
+    //       "price": 50,
+    //       "image": "image1",
+    //     }
+    //     instance.addPart(part, function (error, data, res) {
+    //       if (error) throw error;
+    //       expect(res.status).to.be(200);
+    //       done();
+    //     });
+    //   });
+    //   it('should return bad request', function (done) {
+    //     let part = {
+    //       "model": "test model1",
+    //       "year": 2018,
+    //       "description": "This is test description. All-in-one kit",
+    //       "price": 50,
+    //       "image": "image1",
+    //     }
+    //     instance.addPart(part, function (error, data, res) {
+    //       expect(error.message).to.be("Bad Request");
```

Comment:
You can also check the response status code. Usually the status code is 400 i such cases.